deluge/ui/ui_entry.py

```diff
@@ -32,65 +34,93 @@ def start_ui():
-                       help=_("Sets the default UI to be run when no UI is specified"), action="store")
+    # Setup the initial argument parser. This will be used to change the default
+    # UI as well as setting up the config and logging
+    parser = add_ui_option(BaseArgParser())
 
-    options = parser.parse_args(deluge.common.unicode_argv()[1:])
+    dubious_args = ["-h", "--help", "-v", "-V", "--version"]
+    # Parse arguments without help/version as this is handled later
+    args = [a for a in sys.argv if a not in dubious_args]
+    options = parser.parse_known_ui_args(args)
 
     config = deluge.configmanager.ConfigManager("ui.conf", DEFAULT_PREFS)
     log = logging.getLogger(__name__)
+    log.info("Deluge ui %s", deluge.common.get_version())
 
     if options.default_ui:
         config["default_ui"] = options.default_ui
         config.save()
         log.info("The default UI has been changed to %s", options.default_ui)
         sys.exit(0)
 
-    log.info("Deluge ui %s", deluge.common.get_version())
-    log.debug("options: %s", options)
-
-    selected_ui = options.ui if options.ui else config["default_ui"]
-
-    config.save()
+    default_ui = config["default_ui"]
+    config.save()  # Save in case config didn't already exist
     del config
 
-    # reconstruct arguments to hand off to child client
-    client_args = []
-    if options.args:
-        import shlex
-        client_args.extend(shlex.split(options.args))
+    # We have parsed and got the config dir needed to get the default UI
+    # Now create a parser for choosing the UI.
+    # We need the ui option for parsing to succeed, but result is not used
+    parser = add_ui_option(BaseArgParser(common_help=True))
+
+    # Create a subparser for each registred UI
+    subparsers = parser.add_subparsers(dest='selected_ui', metavar="{%s} [UI args]" % ",".join(ui_titles),
+                                       title=None,  # creates a new argument group with empty title instead
+                                                    # of adding to the positionals group
+                                       help=_("Alternative Deluge UI to launch, with optional args (default UI: *)"))
+    for ui in ui_titles:
+        parser_ui = subparsers.add_parser(ui, common_help=False,
+                                          help=getattr(ui_entrypoints[ui], "cmd_description", ""))
+        parser_ui.add_argument("ui_args", nargs=argparse.REMAINDER)
+        # Change the command name in help to indicate default UI
+        subactions = subparsers._get_subactions()
+        prefix = "*" if ui == default_ui else " "
+        subactions[-1].dest = "%s %s" % (prefix, ui)
+
+    # Handle help which was ignored earlier. If -h/--help option is before the
+    # UI argument, we can now show the help message with a list of possible UIs
+    arg_before_subcmd = parser.set_default_subparser(default_ui, abort_opts=dubious_args)
```

Comment:
You are right. The issue this code handled was actually fixed by the changes here: https://github.com/deluge-torrent/deluge/pull/90/files#diff-1250fcb4c9343ad6037f48af416b41acL83